xwiki-platform-core/xwiki-platform-container/xwiki-platform-container-servlet/src/main/java/org/xwiki/container/servlet/events/SessionDestroyedEvent.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.container.servlet.events;
+
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionEvent;
+
+import org.xwiki.observation.event.Event;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Event triggered when a {@link javax.servlet.http.HttpSession} is destroyed.
+ * This event is the xwiki event triggered when
+ * {@link javax.servlet.http.HttpSessionListener#sessionDestroyed(HttpSessionEvent)} is called.
+ *
+ *  Note that this event should *not* be serializable: we probably don't want it to be sent remotely.
+ *
+ * The following information are also sent:
+ * <ul>
+ *     <li>source: the identifier of the session {@link HttpSession#getId()}</li>
```

Comment:
I really don't see the point, you should simply pass the HttpSession as first argument IMO.

If you really want to do something special with the id you could make it part of the event itself so that someone could listen to the destruction of a specific session for example.

Reply:
Hmmm I'm following a pattern that I saw a lot in XWiki which is to use a String in source of events and an actual object in data of the event. Maybe I imagined it, but I'm usually following this pattern in the event I'm sending.

Reply:
>  a pattern that I saw a lot in XWiki

I cannot think of any (but maybe there is), and I really don't see the point. As I said, that String is more the kind of thing we would put in the event itself so that we can filter on it, but I don't remember any duplicates like this in the parameters.